IdentityServer/v7/docs/content/reference/services/token_creation_service.md

```diff
@@ -0,0 +1,66 @@
+---
+title: "Token Creation Service"
+weight: 50
+---
+
+IdentityServer uses an *ITokenCreationService* which is responsible for the creation
+of tokens, with the default implementation of *DefaultTokenCreationService*.
+
+```csharp
+/// <summary>
+/// Logic for creating security tokens
+/// </summary>
+public interface ITokenCreationService
+{
+    /// <summary>
+    /// Creates a token.
+    /// </summary>
+    /// <param name="token">The token description.</param>
+    /// <returns>A protected and serialized security token</returns>
+    Task<string> CreateTokenAsync(Token token);
+}
+```
+
+The Token creation service involves taking the *Token* model and converting it into
+a JWT. During the JWT creation, you have one last opportunity to
+modify the *Token* by adding, removing, or altering property values. Everyday use cases
+for implementing the *ITokenCreationService* include modifying claims, audiences, and more
+from a secondary data source, such as a profile service, database, or third-party service.
+
+Note that there are better places within IdentityServer's infrastructure to add
+additional claims, such as *IClaimService*, *ITokenService*, and *IProfileService*. We recommend investigating
```

Comment:
A cross reference to https://docs.duendesoftware.com/identityserver/v7/reference/services/profile_service/ would be nice here.